engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnCompensationSubscriptionBehaviour.java

```diff
@@ -72,24 +72,38 @@ public void createCompensationSubscription(
               .setCompensableActivityId(BufferUtil.bufferAsString(context.getElementId()))
               .setCompensableActivityScopeId(
                   BufferUtil.bufferAsString(element.getFlowScope().getId()))
-              .setCompensationHandlerId(compensationHandlerId);
+              .setCompensationHandlerId(compensationHandlerId)
+              .setScopeInstanceKey(context.getFlowScopeKey());
       stateWriter.appendFollowUpEvent(key, CompensationSubscriptionIntent.CREATED, compensation);
     }
   }
 
   public boolean triggerCompensation(
       final ExecutableFlowNode element, final BpmnElementContext context) {
     final Set<CompensationSubscription> subscriptions =
-        compensationSubscriptionState.findSubscriptionsByProcessInstanceKey(
-            context.getTenantId(), context.getProcessInstanceKey());
+        compensationSubscriptionState
+            .findSubscriptionsByProcessInstanceKey(
+                context.getTenantId(), context.getProcessInstanceKey())
+            .stream()
+            .filter(
+                compensationSubscription ->
+                    compensationSubscription.getRecord().getThrowEventId().isEmpty())
+            .collect(Collectors.toSet());
 
     if (subscriptions.isEmpty()) {
       return false;
     }
 
+    final String flowScopeId = BufferUtil.bufferAsString(element.getFlowScope().getId());
+
     final var availableScopeIds =
         getAvailableSubprocessScopeIds((ExecutableFlowElementContainer) element.getFlowScope());
 
+    if (availableScopeIds.isEmpty()) {
+      triggerCompensationForScope(context, subscriptions, flowScopeId, context.getFlowScopeKey());
+      return true;
```

Comment:
:x: We might wait for compensation but without invoking a compensation handler.

Consider the following scenario:

![compensation-outside-of-subprocess](https://github.com/camunda/zeebe/assets/4305769/fc637540-c84d-4ab2-b9ed-a7473be6a6df)

Given:
- The compensation throw event is inside a subprocess
- A task with a compensation handler is outside of the subprocess
- The task is completed

When:
- The compensation throw event is entered

Then:
- No compensation handler is invoked (correct because the handler is outside of the scope)
- The compensation throw event is still active

Expected:
- The compensation throw event is completed because no compensation handler is invoked